feat: Persistent disk, Docker details post, EnvVars in ServicePOST and Image in ServicePOST

tazarov opened the pull request:
- Added support for persistent disk in the WebServiceDetails as it is supported by Render API
- Added an example service with disk support in README.md
- Fixed a typo in the flask example's env type


![image](https://github.com/jackall3n/terraform-provider-render/assets/1157440/4024fcc9-e13b-4f74-936d-df7d4063b95b)

tazarov committed:
feat: Persistent disk support for Web Service Details

tazarov committed:
feat: Persistent disk support for Web Service Details

- Fixed an issue with deleting ServiceEnvironmentResource where API returns 200 but code expects 201
- Added conditional populate of disk struct
- Added `disk` to schema

tazarov commented:
@jackall3n, I tested the changes locally, and everything seems to be working.

jackall3n commented on render/resources/service_environment.go:
```diff
@@ -270,11 +270,11 @@ func (r *serviceEnvironmentResource) Delete(ctx context.Context, req resource.De
 
 	if err != nil {
-		resp.Diagnostics.AddError("failed to update service variables", err.Error())
+		resp.Diagnostics.AddError("failed to update service variables1", err.Error())
```
```suggestion
		resp.Diagnostics.AddError("failed to update service variables", err.Error())
```

jackall3n replied:
Was this a mistake?

hectorj-thetreep replied:
Hi! Is this the only blocking point for this PR?

jackall3n replied:
yeah

tazarov committed:
feat: Persistent disk support for Web Service Details

- Added support for EnvVar in ServicePOST request
- Added support for Image type of service in ServicePOST

tazarov commented:
@jackall3n before this can be merged, you need to merge - https://github.com/jackall3n/render-go/pull/4

jackall3n commented:
> @jackall3n before this can be merged, you need to merge - [jackall3n/render-go#4](https://github.com/jackall3n/render-go/pull/4)

merged
